src/app/bar-chart/bar-chart.component.html

```diff
@@ -0,0 +1,6 @@
+<div class="bar-chart mat-elevation-z2">
+  <canvas baseChart [datasets]="barChartData" [labels]="barChartLabels" [options]="barChartOptions"
```

Comment:
`[labels]="barChartLabels"` 
you don't have this in your code
better use `labels` than `legend`